### PR TITLE
SDIT-1184-KeyWorkerResource

### PR DESCRIPTION
**Pull request by steverendell**

### CODE REVIEW

File: README.MD [hunk 1]
```diff
@@ -153,15 +153,10 @@ docker-compose up -d
...
+Tests can be run individually by adding an e.g. `@wip` tag to the top of the feature file, or to an individual feature scenario, and
+then adding the environment variable `cucumber.filter.tags=@wip` to the executable arguments in 'Edit Confgurations'.
 
-To run all feature tests in IntelliJ, avoiding known @wip or @broken tests, add the following env vars into the 'Edit Configurations' tab:-
-
-| Env var        | Value                         |
-| -------------- | ------------------------------|
-|api.db.target   | nomis                         | 
-|cucumber.options| --tags 'not(@wip or @broken)' |
+To run all feature tests in IntelliJ, avoiding any known @wip or @broken tests, set `cucumber.filter.tags=not(@wip or @broken)`
```
Comment (steverendell) [inline README.MD:159]: Updating cucumber info

File: src/main/java/uk/gov/justice/hmpps/prison/api/resource/KeyWorkerResource.java [hunk 2]
```diff
@@ -39,74 +40,52 @@ public KeyWorkerResource(final KeyWorkerAllocationService keyWorkerService) {
...
-            @ApiResponse(responseCode = "404", description = "Requested resource not found.", content = {@Content(mediaType = "application/json", schema = @Schema(implementation = ErrorResponse.class))}),
-            @ApiResponse(responseCode = "500", description = "Unrecoverable error occurred whilst processing request.", content = {@Content(mediaType = "application/json", schema = @Schema(implementation = ErrorResponse.class))})})
-    @Operation(summary = "Specified key worker's currently assigned offenders.", description = "Specified key worker's currently assigned offenders.")
-    @GetMapping("/{staffId}/agency/{agencyId}/offenders")
-    public List<KeyWorkerAllocationDetail> getAllocationsForKeyworker(@PathVariable("staffId") @Parameter(description = "The key worker staff id", required = true) final Long staffId, @PathVariable("agencyId") @Parameter(description = "The agency (prison) identifier.", required = true) final String agencyId) {
-        return keyWorkerService.getAllocationDetailsForKeyworkers(Collections.singletonList(staffId), agencyId);
-    }
```
Comment (steverendell) [inline src/main/java/uk/gov/justice/hmpps/prison/api/resource/KeyWorkerResource.java:62]: Unused endpoint

File: src/main/java/uk/gov/justice/hmpps/prison/api/resource/KeyWorkerResource.java [hunk 3]
```diff
@@ -39,74 +40,52 @@ public KeyWorkerResource(final KeyWorkerAllocationService keyWorkerService) {
...
-
-    @ApiResponses({
-            @ApiResponse(responseCode = "200", description = "The allocations list is returned.")})
+        @ApiResponse(responseCode = "200", description = "The allocations list is returned.")})
     @Operation(summary = "Retrieves Specified key worker's currently assigned offenders - POST version to allow larger staff lists.", description = "Retrieves Specified key worker's currently assigned offenders - POST version to allow larger staff lists.")
     @PostMapping("/{agencyId}/current-allocations")
+    // @PreAuthorize("hasRole('KEY_WORKER')")
+    @SlowReportQuery
```
Comment (steverendell) [inline src/main/java/uk/gov/justice/hmpps/prison/api/resource/KeyWorkerResource.java:60]: Client (keyworker_api) is currently calling these endpoint with a user/auth token. Leave the @ PreAuthorize until client has switched to a system token

File: src/test/resources/features/key_worker_details.feature [hunk 4]
```diff
@@ -5,41 +5,12 @@ Feature: Key worker details
```
Comment (karenmillermoj) [inline src/test/resources/features/key_worker_details.feature:5]: I deleted the key_worker_available.feature file so it looks as tho the merge has created an empty file (tried to attach the comment to the availble.feature file but it wouldn't let me

Comment (steverendell) [reply src/test/resources/features/key_worker_details.feature:5]: Ah yes i noticed that, result of a merge .. have just pushed update

**Review by lbennett-moj — APPROVED** [review]

**Pull request merged**